learn/guides/ballerina-openapi-support.md

```diff
@@ -0,0 +1,314 @@
+
+Example: 
+```bash
+$ bal openapi -i hello.yaml --mode service
+```
+
+This will generate a Ballerina service for the `hello.yaml` OpenAPI contract named `hello-service` and schemas named types. The above command can be run from anywhere on the execution path. It is not mandatory to run it from within a Ballerina project.
+
+```bash
+The service generation process is complete. The following files were created.
+-- hello-service.bal
+-- types.bal
+```
+#### Generating Ballerina Service by tags
+To generate the Ballerina service stub with a subset of tags defined in the OpenAPI contract, use the options `--tags` and specify the tags you need as specified in the OpenAPI definition.
+```bash
+$ bal openapi -i <openapi contract> [--tags <"tag1","tag2">]
+```
+
+**Example :**
+```bash
+$ bal openapi -i hello.yaml --tags "pets", "list"
+```
+Once you execute the command, only the operations related to given tags will include in generated service file.
+
+**See more command options :** 
+[OpenAPI to Ballerina Command Reference](#openapi-to-ballerina-command-reference)
+
+## Exporting OpenAPI from a Ballerina service (Code-First Approach)
+
+You developer can convert your Ballerina service APIs into a human-readable or machine-readable document such as OpenAPI documents by using the below the Ballerina to OpenAPI command.
+
+
+#### Using the Ballerina to OpenAPI CLI tool
+Export the Ballerina service to an OpenAPI Specification 3.0 definition. For the export to work properly, the input Ballerina service should be defined using the basic service and resource-level HTTP annotations.
+```bash
+$ bal openapi [-i | --input] <ballerina-service-file-path> [(-o | --output) <output-location>]
+```
+Parameter **ballerina-service-file-path** specifies the path of the ballerina service file (e.g., `my-api.bal`) and is mandatory.
+If your Ballerina file include multiple services, this command generates OpenAPI contract for each service in the Ballerina file.
+
+#### Generating OpenAPI Specification with `json` Format
+Use `--json` flag If you need the Ballerina service to OpenAPI output as JSON. The default is YAML.
+```bash
+$ bal openapi -i <ballerina resource file> [--json]
+```
+#### Generating OpenAPI Specification for Given Service
+If you need to document an OpenAPI contract for only one given service, then use this command. Use service name as the service `absolute-resource-path`.
+
+```bash
+$ bal openapi -i <ballerina resource file> [-s|--service] <service-name>
+```
+**Example :**
+```bash
+$ bal openapi -i helloService.bal
+```
+This will generate the OpenAPI contracts for the Ballerina services, which are in the `helloService.bal` Ballerina file.
+
+```bash
+$ bal openapi -i helloService.bal -s "/hello"
+```
+
+This will generate the OpenAPI contracts for the Ballerina service which absolute-resource-path is `/hello`, which are in the `helloService.bal` Ballerina file.
```

Comment:
```suggestion
This will generate the OpenAPI contracts for the Ballerina service of which the `absolute-resource-path` is `/hello`, which are in the `helloService.bal` Ballerina file.
```